spec/models/user_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe User do
+  subject(:user) { build(:user) }
+  let(:plain_user) { create(:user) }
+
+  it_behaves_like 'a Hyrax User'
+
+  it "has a display name" do
+    expect(user.display_name).not_to be_empty
+  end
+
+  it '#to_s yeilds the #display_name' do
+    expect(user.to_s).to eq user.display_name
+  end
+
+  describe 'roles' do
+    it 'is emplty for a new user' do
+      new_user = create(:user)
+      expect(new_user.roles).to be_empty
+    end
+
+    it "#add_role adds a named role" do
+      plain_user.add_role('some_role')
+      expect(plain_user.roles.inspect).to match(/some_role/)
+    end
+
+    it "#add_role creates a new role if needed" do
+      expect { plain_user.add_role('named_role') }.to change { Role.count }.by(1)
+    end
+
+    it "#add_role uses an existing role if it exists" do
+      role = Role.create(name: 'existing_role')
+      plain_user.add_role('existing_role')
```

Comment:
`.to change` here? This could pass if `'existing_role'` was somehow added in setup.

Reply:
👍